### PR TITLE
MEGNet Dataloader class is added

### PR DESCRIPTION
**Pull request by kenko911**

### CODE REVIEW

File: megnet/dataloader/MEGNetDataset.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        Check if the dgl_graph.bin exists or not
+        """
+        graph_path = str(self.path) + "/dgl_graph.bin"
+        return os.path.exists(graph_path)
+
+    def process(self):
+        """
+        Call _load_graph to convert Pymatgen structure into dgl graphs
+        """
+        self.graphs = self._load_graph()
```
Comment (shyuep) [inline megnet/dataloader/MEGNetDataset.py:93]: This is unnecessary? Why not just put the code from self._load_graph() in the process method?

File: megnet/dataloader/MEGNetDataset.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        num_graphs = self.labels.shape[0]
+        graphs = []
+
+        for idx in trange(num_graphs):
+            structure = self.structures[idx]
+            graph, state_attr = self.cry_graph.get_graph_from_structure(structure=structure)
+            graphs.append(graph)
+        return graphs
+
+    def save(self):
```
Comment (shyuep) [inline megnet/dataloader/MEGNetDataset.py:108]: I think you can provide a filename argument to allow people to specify the filename. The default can be `dgl_graph.bin`.

File: megnet/dataloader/MEGNetDataset.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        """
+        Save dgl graphs
+        """
+        graph_path = str(self.path) + "/dgl_graph.bin"
+        labels_with_key = {self.label_name: self.labels}
+        save_graphs(graph_path, self.graphs, labels_with_key)
+
+    def load(self):
```
Comment (shyuep) [inline megnet/dataloader/MEGNetDataset.py:116]: Similarly, load can take a filename.

File: megnet/dataloader/MEGNetDataset.py (hunk 1 above)
Comment (kenko911) [reply megnet/dataloader/MEGNetDataset.py:93]: Hi, Shyue Ping, I just tried to put the code in the process method and it doesn't work in the expected way meaning that the length of training, validation and testing dataloader are incorrect. I am trying to figure it out why but I believe there maybe some operations done in DGLDataset class.

File: megnet/dataloader/MEGNetDataset.py (hunk 2 above)
Comment (kenko911) [reply megnet/dataloader/MEGNetDataset.py:108]: Yup, I can add that

File: megnet/dataloader/MEGNetDataset.py (hunk 3 above)
Comment (kenko911) [reply megnet/dataloader/MEGNetDataset.py:116]: Yup, I can add that.

File: megnet/dataloader/MEGNetDataset.py (hunk 1 above)
Comment (shyuep) [reply megnet/dataloader/MEGNetDataset.py:93]: I don't think this is very likely. The process method is a one-liner that sets the self.graph.

File: megnet/dataloader/MEGNetDataset.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        structures: Pymatgen strutcure
+        labels: property values
+        label: label name
+        cry_graph: Pmg2Graph
+        """
+        self.cry_graph = cry_graph
+        self.structures = structures
+        self.labels = torch.FloatTensor(labels)
+        self.label_name = label_name
+        self.path = os.getcwd()
```
Comment (shyuep) [inline megnet/dataloader/MEGNetDataset.py:79]: This is not necessary. All paths are by default from the current working directory. You do not need this variable.

**Pull request merged**